using_images/db_images/mongodb.adoc

```diff
@@ -134,3 +134,278 @@ The MongoDB image can be run with mounted volumes to enable persistent storage f
+`rs0` by default.
+
+==== Creating the service pod
+
+To initialize members created by the DeploymentConfig a "service pod" has to be
+defined in the Template. This pod will start MongoDB with `initiate` argument,
+which will make the container entrypoint behave slighly different than having a
+regular standalone MongoDB database.
+
+The service pod defined in the example replication template looks like this:
+
+[source,json]
+----
+{
+  "kind": "Pod",
+  "apiVersion": "v1beta3",
+  "metadata": {
+    "name": "mongodb-service",
+    "creationTimestamp": null,
+    "labels": {
+      "name": "mongodb-service"
+    }
+  },
+  "spec": {
+    "restartPolicy": "Never",
+    "dnsPolicy": "ClusterFirst",
+    "containers": [
+      {
+        "name": "initiate",
+        "image": "openshift/mongodb-24-centos7",
+        "args": ["initiate"],
+        "env": [
+          {
+            "name": "MONGODB_USER",
+            "value": "${MONGODB_USER}"
+          },
+          {
+            "name": "MONGODB_PASSWORD",
+            "value": "${MONGODB_PASSWORD}"
+          },
+          {
+            "name": "MONGODB_DATABASE",
+            "value": "${MONGODB_DATABASE}"
+          },
+          {
+            "name": "MONGODB_ADMIN_PASSWORD",
+            "value": "${MONGODB_ADMIN_PASSWORD}"
+          },
+          {
+            "name": "MONGODB_REPLICA_NAME",
+            "value": "${MONGODB_REPLICA_NAME}"
+          },
+          {
+            "name": "MONGODB_SERVICE_NAME",
+            "value": "${MONGODB_SERVICE_NAME}"
+          },
+          {
+            "name": "MONGODB_KEYFILE_VALUE",
+            "value": "${MONGODB_KEYFILE_VALUE}"
+          }
+        ]
+      }
+    ]
+  }
+}
+----
+
+==== Creating headless service
+
+The `initiate` argument in container specification above will make the container
+first discover all running member pods within the MongoDB cluster. To achieve
+that, we defined a 'headless' service named 'mongodb' in the example template.
+
+To have a 'headless' service, you have to set the `portalIP` field value in the
+service definition to `None`.  Then you can use a DNS query to get a list of the
+pod IP addresses that represents the current endpoints for this service.
+
+The example replication template defines the headless service as following:
+
+[source,json]
+----
+{
+  "kind": "Service",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "${MONGODB_SERVICE_NAME}",
+    "labels": {
+      "name": "${MONGODB_SERVICE_NAME}"
+    }
+  },
+  "spec": {
+    "ports": [
+      {
+        "protocol": "TCP",
+        "port": 27017,
+        "targetPort": 27017,
+        "nodePort": 0
+      }
+    ],
+    "selector": {
+      "name": "mongodb-replica"
+    },
+    "portalIP": "None",
+    "type": "ClusterIP",
+    "sessionAffinity": "None"
+  },
+  "status": {
+    "loadBalancer": {}
+  }
+}
+----
+
+==== Creating the final replication set
+
+When the entrypoint has the IP addresses of all running MongoDB members, it
```

Comment:
not clear what "when the entrypoint has the ip addresses" means... what's the entrypoint?

I assume you mean the script that is run as the image's entrypoint, but it's not obvious.